tests/function_helpers.go

```diff
@@ -1,52 +1,47 @@
 package tests
 
 import (
-	"encoding/json"
+	"context"
 	"net/http"
 	"path"
 	"testing"
 
+	faasSDK "github.com/openfaas/faas-cli/proxy"
 	"github.com/openfaas/faas-provider/types"
 )
 
-func deploy(t *testing.T, createRequest types.FunctionDeployment) int {
+func deploy(t *testing.T, createRequest *faasSDK.DeployFunctionSpec) int {
 	t.Helper()
+	gwURL := gatewayUrl(t, "", "")
```

Comment:
It seems like all of these uses are now the same `gatewayUrl(t, "", "")` perhaps we should just refactor it to being `gwURL := gatewayURL(t)`

Reply:
It is still being used at some places. I will create one more function which will read env variable and use it inside `gatewayUrl`. We can use the new function to just read the gateway URL.